Purchases/Purchasing/PurchasesOrchestrator.swift

```diff
@@ -233,11 +234,13 @@ class PurchasesOrchestrator {
 
     @available(iOS 12.2, macOS 10.14.4, watchOS 6.2, macCatalyst 13.0, tvOS 12.2, *)
     func purchase(sk1Product: SK1Product,
-                  storeProductDiscount: StoreProductDiscountType,
+                  discount: StoreProductDiscountType,
                   package: Package?,
                   completion: @escaping PurchaseCompletedBlock) {
-        self.promotionalOffer(forProductDiscount: storeProductDiscount,
-                              product: sk1Product) { [unowned self] promotionalOffer, error in
+        self.promotionalOffer(
+            forProductDiscount: discount,
+            product: StoreProduct(sk1Product: sk1Product)
+        ) { [unowned self] promotionalOffer, error in
```

Comment:
🤔 
I'm slightly concerned about the `unowned` usage here: 
- You're not _supposed_ to call configure multiple times, but in practice, people do
- If you do call configure multiple times, the old instance is set to nil and the new one is the shared instance. 
- If this happened and the `promotionalOffer(forProductDiscount:product:)` method returns, this will hard crash. 

Perhaps a `weak self` + early return here would be better?

Reply:
Actually, this doesn't need `unowned` or `weak`. The lifetime of the closure is limited, so what we really want is to call `completion` when this finishes. There aren't retain cycles here.